utils/_context/containers.py

```diff
@@ -473,3 +481,34 @@ def __init__(self, host_log_folder) -> None:
             host_log_folder=host_log_folder,
             healthcheck={"test": "/healthcheck.sh", "retries": 60},
         )
+
+
+class OpenTelemetryCollectorContainer(TestedContainer):
+    def __init__(self, host_log_folder) -> None:
+        super().__init__(
+            image_name="otel/opentelemetry-collector-contrib:latest",
+            name="collector",
+            command="--config=/etc/otelcol-config.yml",
+            environment={
+                "DD_API_KEY": os.environ.get("DD_API_KEY_3"),
+                "DD_SITE": os.environ.get("DD_SITE", "datad0g.com"),
+            },
+            volumes={"./utils/build/docker/otelcol-config.yaml": {"bind": "/etc/otelcol-config.yml", "mode": "ro",}},
+            host_log_folder=host_log_folder,
+            ports={"13133/tcp": ("0.0.0.0", 13133)},
+        )
+
+    # Override wait_for_health because we cannot do docker exec for container opentelemetry-collector-contrib
+    def wait_for_health(self):
```

Comment:
Why not using the healthcheck of `TestedContainer` ?

Reply:
We cannot exec any commands for container `opentelemetry-collector-contrib`, e.g. even echo / cd / ls don't work.